myaws/acm_ls.go

```diff
@@ -0,0 +1,89 @@
+package myaws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/acm"
+	"github.com/pkg/errors"
+)
+
+// ACMLsOptions customize the behavior of the Ls command.
+type ACMLsOptions struct {
+	Verbose bool
+	Pending bool
+}
+
+// ACMLs describes Amazon Certificate resources.
+func (client *Client) ACMLs(options ACMLsOptions) error {
+	certificates, err := client.listsCertificate()
+	if err != nil {
+		return errors.Wrap(err, "listsCertificate failed:")
+	}
+	for _, cert := range certificates {
+		if options.Pending && !haveAnyPending(cert) {
+			continue
+		}
+
+		if options.Verbose {
+			fmt.Fprintln(client.stdout, cert.GoString())
+		} else {
+			fmt.Fprintln(client.stdout,
+				cert.NotAfter,
+				*cert.DomainName,
+			)
+		}
+	}
+
+	return nil
+}
+
+func (client *Client) listsCertificate() ([]*acm.CertificateDetail, error) {
+	var certificates []*acm.CertificateDetail
+	params := &acm.ListCertificatesInput{}
+
+	response, err := client.ACM.ListCertificates(params)
+	if err != nil {
+		return certificates, errors.Wrap(err, "ListCertificates failed:")
+	}
+
+	for {
+		for _, cert := range response.CertificateSummaryList {
+			describeParams := &acm.DescribeCertificateInput{
+				CertificateArn: cert.CertificateArn,
+			}
+			r, err := client.ACM.DescribeCertificate(describeParams)
+			if err != nil {
+				return certificates, errors.Wrap(err, "DescribeCertificate failed:")
+			}
+			certificates = append(certificates, r.Certificate)
+		}
+
+		if response.NextToken == nil {
+			break
+		}
+
+		params.SetNextToken(*response.NextToken)
+		response, err = client.ACM.ListCertificates(params)
+		if err != nil {
+			return certificates, errors.Wrap(err, "ListCertificates failed:")
+		}
+	}
+
+	return certificates, nil
+}
+
+func haveAnyPending(certificate *acm.CertificateDetail) bool {
```

Comment:
I prefer `hasAnyPendingACMCertificate()` instead of `haveAnyPending()`.
Because the namespace of the current `myaws` package is not structured.